# Configuration Settings for the Rover Rx

# INS Parameters

device: tcp://192.168.3.1:28784

serial:
  baudrate: 921600
  hw_flow_control: "off"

stream_device:
  tcp:
    ip_server: ""
    port: 0
  udp:
    ip_server: ""
    port: 0
    unicast_ip: ""

configure_rx: true

login:
  user: ""
  password: ""

osnma:
  mode: "loose"
  ntp_server: ""
  keep_open: true

frame_id: gnss

imu_frame_id: imu

poi_frame_id: base_link

vsm_frame_id: vsm

aux1_frame_id: aux1

vehicle_frame_id: base_link

local_frame_id: odom

insert_local_frame: false

get_spatial_config_from_tf: true

lock_utm_zone: true

use_ros_axis_orientation: true

receiver_type: ins

multi_antenna: true

datum: Default

att_offset:
  heading: 0.0
  pitch: 0.0

ant_type: "Unknown"
ant_serial_nr: "Unknown"
ant_aux1_type: "Unknown"
ant_aux1_serial_nr: "Unknown"

polling_period:
  pvt: 0
  rest: 500

use_gnss_time: false

latency_compensation: false

rtk_settings:
  keep_open: true
  ntrip_1:
    id: ""
    caster: ""
    caster_port: 2101
    username: ""
    password: ""
    mountpoint: ""
    version: "v2"
    tls: false
    fingerprint: ""
    rtk_standard: "auto"
    send_gga: "auto"
  ip_server_1:
    id: ""
    port: 0
    rtk_standard: "auto"
    send_gga: "auto"
  serial_1:
    port: ""
    baud_rate: 115200
    rtk_standard: "auto"
    send_gga: "auto"

publish:
  # For both GNSS and INS Rxs
  navsatfix: false
  gpsfix: false
  gpgga: false
  gprmc: false
  gpst: false
  measepoch: false
  pvtcartesian: false
  pvtgeodetic: false
  basevectorcart: false
  basevectorgeod: false
  poscovcartesian: false
  poscovgeodetic: false
  velcovcartesian: false
  velcovgeodetic: false
  atteuler: false
  attcoveuler: false
  pose: false
  twist: true
  diagnostics: true
  aimplusstatus: true
  galauthstatus: false
  # For INS Rx only
  insnavcart: true
  insnavgeod: true
  extsensormeas: true
  imusetup: false
  velsensorsetup: false
  exteventinsnavcart: false
  exteventinsnavgeod: false
  imu: true
  localization: true
  tf: true
  localization_ecef: false
  tf_ecef: false


# INS-Specific Parameters

ins_spatial_config:
  imu_orientation:
    theta_x: 0.0
    theta_y: 0.0
    theta_z: 0.0
  poi_lever_arm:
    delta_x: 0.0
    delta_y: 0.0
    delta_z: 0.0
  ant_lever_arm:
    x: 0.0
    y: 0.0
    z: 0.0
  vsm_lever_arm:
    vsm_x: 0.0
    vsm_y: 0.0
    vsm_z: 0.0

ins_initial_heading: auto

ins_std_dev_mask:
  att_std_dev: 5.0
  pos_std_dev: 10.0

ins_use_poi: true

ins_vsm:
  ros:
    source: ""
    config: [false, false, false]
    variances_by_parameter: false
    variances: [0.0, 0.0, 0.0]
  ip_server:
    id: ""
    port: 0
    keep_open: true
  serial:
    port: ""
    baud_rate: 115200
    keep_open: true
  
# logger

activate_debug_log: false